public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx

```diff
@@ -123,6 +132,20 @@ export default class UpdateAlertConditions extends Component<
     }
   };
 
+  async getPlugins() {
```

Comment:
This is same method as above, let's make this a utility function in a helper file and use that?

Reply:
Yes, I was also thinking on it.

It is the same function on both places (here and in `Findings.tsx`), but result is written to the component state object which I would not force to be named in the same way (`{...plugins: string[];}`).

There are 2 points worth mentioning/discussing:

1. This could be interesting if we would go with functional components and create an custom hook like this:
  ```
  import { usePlugins } from "helperHooks";
  const { plugins } = usePlugins();
  ```
  of corse we would need to think more on different use cases, but generally it could be more "react team" practices aligned. This is just initial point to talk, I'm also not hard opined on this topic yet.

2. I saw in other plugin (index management) too, that frontend services are forwarding whole response to the "consumer", and then consumer (in this case `<UpdateAlertConditions>`) deals with `response.ok` flag. I guess there is a reason for this, but I still haven't got it. I would move that `if(response.ok)` to the service and throw and "native" js error in the else branch. Consumer would just have try/catch around the service call. Ex:
```
// service.ts
...
getPlugins = async (): Promise<ServerResponse<Plugin[]>> => {
 let url = `..${API.PLUGINS}/`;
 const responseObject = await this.httpClient.get(url);

 if (resposne.ok) {
   return responseObject.response;
 } else {
    throw resposne.error;
 }
};
...

// component.tsx
...
try {
  const plugins = await opensearchService.getPlugins();
  this.setState({ plugins: plugins.map((plugin) => plugin.component) });
} catch (e) {
  errorNotificationToast(notifications, 'retrieve', 'plugins', e);
}
...
```

Also, not hardly opined on this just adding this discussion as a contribution.

Let me know if you had some other idea how to move this to the helper by current convenience.

Reply:
As discussed, let's extract out the common method and have it return the plugins array (which will be empty in case of error). I will refactor it to abstract out the check for "ok" in the service files itself. Thank you for that suggestion

Reply:
Makes sense, will incorporate this feedback in the refactor